Order by alias example in Java section (#229)
cap-js/docs: java/query-api.md

    .orderBy(c -> c.get("ID").desc(), c -> c.get("title").asc());
```

You can order by the alias of a column of the select list or a column that is defined as a result of the function call. 

```java
Select.from("bookshop.Person")
    .columns(p -> p.get("name").toUpper().as("aliasForName"))
    .orderBy(p -> p.get("aliasForName").asc());
```

> Aliases of columns have precedence over the element names when `orderBy` is evaluated. Aliases may shadow the respective elements. To avoid that shadowing, don't use aliases that equal existing element names.

Reviewer: ```suggestion
Aliases of columns have precedence over the element names when `orderBy` is evaluated. 

::: warning
Aliases may shadow elements names. To avoid shadowing, don't use element names as aliases.
::::
```
